src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -151,13 +153,32 @@ public Object instanceMainPostParseArgs() {
 
         LoggingUtils.setLoggingLevel(VERBOSITY);  // propagate the VERBOSITY level to logging frameworks
 
-        for (final File f : TMP_DIR) {
+        for (final String s : TMP_DIR) {
+            final Path p = IOUtils.getPath(s);
             // Intentionally not checking the return values, because it may be that the program does not
             // need a tmp_dir. If this fails, the problem will be discovered downstream.
-            if (!f.exists()) f.mkdirs();
-            f.setReadable(true, false);
-            f.setWritable(true, false);
-            System.setProperty("java.io.tmpdir", f.getAbsolutePath()); // in loop so that last one takes effect
+            if (!Files.exists(p)) {
+                try {
+                    Files.createDirectories(p);
+                } catch (IOException e) {
+                    // intentionally ignoring
+                }
+            }
+            try {
+                // add the READ/WRITE permissions to the actual permissions of the file
+                final Set<PosixFilePermission> permissions = Files.getPosixFilePermissions(p);
+                // only set the permissions if they change
+                if (permissions.addAll(Arrays.asList(
+                        PosixFilePermission.OWNER_READ, PosixFilePermission.GROUP_READ, PosixFilePermission.OTHERS_READ,
+                        PosixFilePermission.OWNER_WRITE, PosixFilePermission.GROUP_WRITE, PosixFilePermission.OTHERS_WRITE))) {
```

Comment:
By the way, this is related with https://github.com/broadinstitute/gatk/issues/4513 - should probably resolve here.